include/seqan3/alphabet/concept.hpp

```diff
@@ -678,50 +678,107 @@ inline constexpr auto alphabet_size_v = detail::adl::only::alphabet_size_fn<alph
+ * by value.
  *
- * Types that satisfy the concept (and all refinements) can be serialised via SeqAn3
- * serialisation support.
- * The rank value is (de-)serialised, types need not provide any overloads themselves.
+ * It is highly recommended that non-reference types that model this concept, also model:
+ *
+ *   * std::Regular
+ *   * std::is_trivially_copyable
+ *   * seqan3::StandardLayout
  *
- * \par Concepts and doxygen
+ * All alphabets available in SeqAn (with very few exceptions) do so.
  *
- * The requirements for this concept are given as related functions and metafunctions.
- * Types that satisfy this concept are shown as "implementing this interface".
+ * ### Related types
+ *
+ * If a given type `t` models this concept, the following types typically do so, as well:
+ *
+ *   * `t &`
+ *   * `t const`
+ *   * `t const &`
  */
 //!\cond
 template <typename t>
-SEQAN3_CONCEPT Semialphabet = std::StrictTotallyOrdered<t> &&
-                                requires (t v)
+SEQAN3_CONCEPT Semialphabet =
+    std::StrictTotallyOrdered<t> &&
+    std::CopyConstructible<t> &&
+    std::is_nothrow_copy_constructible_v<t> &&
+    requires (t v)
 {
-    // static data members
-    alphabet_size_v<t>;
+    requires seqan3::alphabet_size_v<t> >= 0;
 
-    // conversion to rank
-    requires           noexcept(to_rank(v));
+    { seqan3::to_rank(v) };
+};
+//!\endcond
+
+// ============================================================================
+// WritableSemialphabet
+// ============================================================================
 
-    // assignment from rank
-    requires           noexcept(assign_rank_to(0, v                           ));
-    requires std::Same<decltype(assign_rank_to(0, v                           )), std::remove_reference_t<t> &>;
+/*!\interface seqan3::WritableSemialphabet <>
+ * \brief A refinement of seqan3::Semialphabet that adds assignability.
+ * \extends seqan3::Semialphabet
+ * \ingroup alphabet
+ *
+ * This concept refines seqan3::Semialphabet and adds the requirement to be able to change the value by
+ * assigning a value of the rank representation.
+ *
+ * For a detailed overview of how the different alphabet concepts are related, see
+ * \ref alphabet module.
+ *
+ * ### Concepts and doxygen
+ *
+ * Concepts are shown as "interfaces" in our documentation for technical reasons.
+ * Types that model this concept are shown as "implementing this interface".
+ *
+ * ### Requirements
+ *
+ *   1. `t` shall model seqan3::Semialphabet
+ *   2. seqan3::assign_rank_to needs to be defined for objects of type `t`
+ *
+ * See the documentation pages for the respective requirements.
+ *
+ * ### Related types
+ *
+ * If a given type `t` models this concept, the following types typically do so, as well:
+ *
+ *   * `t &`
+ *
+ * `const`-qualified types on the other hand are not assignable.
+ *
+ * ### Serialisation
+ *
+ * Types that model the concept (and all refinements) can be serialised via SeqAn3
```

Comment:
At some places you refer to SeqAn at others to SeqAn3. And I stumbled over this by myself quite often. We need to decide whether we refer to SeqAn (no matter what the version is) or are version agnostic. I am tending more to not include the version. It is still SeqAn, and the documentation is versioned together with the source code, so it is clear which version it is referring to. But if we have at some point SeqAn4 which is likely to not be a complete overhaul of SeqAn3 then it is weird to read that some entities are documented referring to SeqAn3.

Reply:
> At some places you refer to SeqAn at others to SeqAn3. And I stumbled over this by myself quite often. We need to decide whether we refer to SeqAn (no matter what the version is) or are version agnostic. I am tending more to not include the version.

Yep, that's a good point. I usually had the policy to just write "SeqAn" and write "SeqAn3" only to highlight a difference to SeqAn2. But I guess we should just have a "Porting Guide" for that.

Reply:
Can you then replace the SeqAn3 as discussed.